service/controller/v5/resource/instance/status.go

```diff
@@ -160,3 +161,19 @@ func resourceStatusExists(customObject providerv1alpha1.AzureConfig, t string) b
 
 	return false
 }
+
+// unversionedName provides a comparable name without the exact version number
+// suffix. When the resource name should ever change, the mechanism of managing
+// the resource status has to be adapted accordingly. Otherwise unversionedName
+// will return an empty string and the mechanism will break in unexpected ways.
+func unversionedName(name string) string {
+	if !strings.HasPrefix(name, "instancev") {
+		return ""
+	}
+
+	return "instancev"
```

Comment:
I think it would be good to drop the `v4` thing at all. I doesn't fit really well versioning of operator roadmap. When this is in place I think the resource will be just called `instance`. What I thought long term might make sense is adding `version` and `controller` to the log context in `operatorkit` instead.

Reply:
Agreed. This code here is the perfect transitioning tool for it. We can then just change the name and once migrated remove the obsolete magic.

Reply:
Can't we change the value of the `status` to `instance` right away?